vue/components/ui/molecules/select/select.vue

```diff
@@ -263,11 +313,19 @@ export const Select = {
         openDropdown() {
             if (this.disabled || this.visibleData) return;
             this.visibleData = true;
+
+            // focus input and with blank text after dropdown is opened
+            if (this.filter) {
+                this.filterText = "";
+                this.$nextTick(() => this.$refs.input.focus());
+            }
         },
-        closeDropdown() {
+        closeDropdown(focus = true) {
             if (!this.visibleData) return;
             this.dehighlight();
             this.visibleData = false;
+            // focus the select button after dropdown is closed
+            if (this.filter && focus) this.$nextTick(() => this.$refs.selectButton.focus());

```

Comment:
It seems weird that I can ask `closeDropdown(focus = true)` and then no focus happens because I don't have a `flter`. Having this said, you can achieve the same result and keep this function sane by moving the `this.filter` logic to the caller of this function

Reply:
Focus is already happening by default if I don't have a `filter`. This is to fix the behavior of the filter because it would focus an element that would hide right after.

But there indeed a problem with this because when I do `closeDropdown(focus = false)` it will in fact focus in non-filter mode.

Reply:
refactoring..

Reply:
thinking better, there are a lot of places (8 places) that call closeDropdown and want `focus = false`
maybe it's better to just remove focus arg? as we will probably always want to focus the selectButton after closeDropdown

Reply:
If you can always call focus without any issue then yes, I'd do that.

Reply:
I renamed to `maintainFocus` as removing the arg would stop global click from unfocusing.

Focus will always maintain itself in select for filter and non-filter mode.
The rename gives better info that if focus wasn't there in the beginning, it will not focus.